stream.go

```diff
@@ -132,6 +136,43 @@ func (s *Stream) ReadSCTP(p []byte) (int, PayloadProtocolIdentifier, error) {
 	}
 }
 
+// SetReadDeadline sets the read deadline in an identical way to net.Conn
+func (s *Stream) SetReadDeadline(deadline time.Time) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if s.readTimeoutCancel != nil {
+		close(s.readTimeoutCancel)
+		s.readTimeoutCancel = nil
+	}
+
+	if deadline.IsZero() {
```

Comment:
I feel if s.readErr != nil && s.readErr != errReadDeadlineExceeded, we could just return (probably with nil).

Reply:
Example:

```go
// SetReadDeadline sets the read deadline in an identical way to net.Conn
func (s *Stream) SetReadDeadline(deadline time.Time) error {
	s.lock.Lock()
	defer s.lock.Unlock()

	if s.readTimeoutCancel != nil {
		close(s.readTimeoutCancel)
		s.readTimeoutCancel = nil
	}

	if s.readErr != nil {
		if !errors.Is(s.readErr, errReadDeadlineExceeded) {
			return nil	
		}
		s.readErr = nil
	}

	s.readTimeoutCancel = make(chan struct{})

	go func(readTimeoutCancel chan struct{}) {
		// as-is
	}
	return nil
}
```

Reply:
Agreed, added